packages/manager/apps/pci-block-storage/src/pages/new/New.page.tsx

```diff
@@ -65,19 +70,36 @@ export default function NewPage(): JSX.Element {
         true,
       );
     },
-    onError: (err: ApiError) => {
+    onError: (err: unknown) => {
       stepper.validation.step.unlock();
-      addError(
-        <Translation ns="add">
-          {(tr) =>
-            tr('pci_projects_project_storages_blocks_add_error_post', {
-              volume: stepper.form.volumeName,
-              message: err?.response?.data?.message || err?.message || null,
-            })
-          }
-        </Translation>,
-        true,
-      );
+      if (isApiCustomError(err)) {
+        if (isMaxQuotaReachedError(err)) {
+          addError(
+            <Trans
+              ns="add"
+              i18nKey="pci_projects_project_storages_blocks_add_error_quota_exceeded"
+              components={{
+                Link: (
+                  <ExternalLink href={`${projectUrl}/quota`} isTargetBlank />
+                ),
+              }}
+            />,
+            true,
+          );
+        } else {
+          addError(
+            <Trans
+              ns="add"
+              i18nKey="pci_projects_project_storages_blocks_add_error_post"
+              values={{
+                volume: stepper.form.volumeName,
+                message: err?.response?.data?.message || err?.message || null,
```

Comment:
Why so many unnecessary optional chainings? We should constraint the type returned by `isApiCustomError` type guard.